features/verification.py

```diff
@@ -63,7 +61,7 @@ async def gen_code_and_send_mail(
             code = "".join(random.choices(string.ascii_uppercase + string.digits, k=6))
             mail_content = utils.fill_message("verify_mail_content", code=code)
             # Save the newly generated code into the database
-            self.repo.save_sent_code(user.login, code)
+            user.save_sent_code(user.login, code)
```

Comment:
No need to send login, you already have it in user instance. Just convert the method from classmethod to regular one.
```suggestion
            user.save_sent_code(code)
```